### DISCUSSION TITLE
Python #| fig-cap:  produces <p>  tags not <figcaption>

### DISCUSSION THREAD

**Discussion by astrowonk** [quarto-dev/quarto-cli#7994] [Q&A] [answered]
### Description

````qmd

```{python}
#| echo: false
#| fig-cap: "This is my figure caption"

fig =  makeplotlyfig() #some function that makes a plotly figure
fig.show()
```




````

So, the above code in Quarto 1.4x, there is a caption underneath the plotly figure but it is wrapped in `<p>` tags and looks exactly like the rest of the text. By contrast, a Figure caption added to an inline image ala `![Elephant](elephant.png)` the caption is wrapped in `<figcaption>`.

I think this is bug? EDITED to say that this is Quarto version `1.4.532`

**Comment by cscheid** [comment]
We use `figcaption` for _floats_ that have a caption. It's not great to emit `<p>`, but it's not exactly a bug; `fig-cap` by itself doesn't make a _float_, because it's lacking the "fig-" or "tbl-" label.

(`![Elephant](elephant.png)` gets its own treatment because that's a Figure according to Pandoc.)

Can you give me a full .qmd with the contents of `makeplotlyfig()`? I don't want to assume and make this even more confusing.

**Reply by astrowonk** [reply]
I made a MRE to show what's happening. Maybe it's by design but captions that looks like paragraphs don't distinguish themselves in any way from the next paragraph, so for me they don't have much utility.

````qmd
---
title: "What's up with fig-cap?"
format: html
jupyter: python3

---


```{python}
#| fig-cap: "This is my figure caption"

import plotly.express as px
df = px.data.iris() # iris is a pandas DataFrame
px.scatter(df, x="sepal_width", y="sepal_length")

```

The caption above looks *exactly the same as the text in this paragraph.* Both `<p>` tags so that's expected!

Whereas:

![Graph of Figure](https://upload.wikimedia.org/wikipedia/commons/7/76/Polynomial_of_degree_three.svg){width=300}

Has a nice `<figcaption>` that makes the caption look different than the next paragraph.

Matplotlib gets figcaptions:

```{python}
#| fig-cap: "Line Plot 1"

import matplotlib.pyplot as plt
plt.plot([1,23,2,4])
plt.show()

```
Above is a nice `<figcaption>` wrapped fig caption of  "Line Plot 1"

````

There's nothing unusual about the block that makes the plotly figure, so this simple example works. Edited to show that matplotlib example code in the Quarto figures documentations does get `<figcaption>` treatment.

**Reply by astrowonk** [reply]
I can do something of a workaround….

````qmd

::: {#fig-graph}

```{python}
#| echo: false

df = px.data.iris() # iris is a pandas DataFrame
px.scatter(df, x="sepal_width", y="sepal_length")

```
My scatter plot.
:::
````

but then the caption is prepended with `Figure 1`.

**Reply by cscheid** [reply]
Ok, I take it all back. In `r` code cells, we do the right thing. This is a bug!

**Reply by cscheid** [reply, marked as answer]
Tracking in #7999.
